lib/mix/lib/mix/tasks/deps.ex

```diff
@@ -120,6 +120,10 @@ defmodule Mix.Tasks.Deps do
     * `:subdir` - (since v1.13.0) search for the project in the given directory
       relative to the git checkout. This is similar to `:sparse` option but instead
       of a doing a sparse checkout it does a full checkout.
+    * `:depth` - (since v1.17.0) creates a shallow clone of the Git repository,
```

Comment:
Bringing back the original note:

> This will require an update to the version depending on when this would be released.

@josevalim do you prefer to hold this off until 1.17 or 1.16/1.16.x?